src/deadline/client/ui/dialogs/submit_job_to_deadline_dialog.py

```diff
@@ -142,6 +143,15 @@ def __init__(
         self.gui_update_counter: Any = None
         self.refresh_deadline_settings()
 
+    def _submission_succeeded_signal_receiver(self, job_id: str):
+        self.job_id = job_id
+
+        set_setting("defaults.job_id", job_id)
+
+    def _close_event_receiver(self):
+        if self.submitter_name != "JobBundle" and self.job_id:
```

Comment:
You've maintained the existing behavior intent here, but closing or not depending on the value of submitter_name feels like a design choice asking for improvement. Maybe in a future change we could change this to a parameter to the dialog like "close_on_complete" instead, and then have each caller make the choice explicitly.

Reply:
Agreed. I'll cut an issue to capture this